.travis.yml

```diff
@@ -205,16 +204,23 @@ after_success:
     git tag master;
     ./gradlew uploadArchives; 
   fi;
-
+- if [[ $TRAVIS_SECURE_ENV_VARS == true && $TRAVIS_EVENT_TYPE != cron ]]; then
+    if [[ ${TESTS_REQUIRE_GCLOUD} != "true" ]]; then
```

Comment:
We should just make this script do nothing if the authentication already exists. This feels like a point of confusion in the script that will cause problems if we change things later.

Reply:
That's a good idea.  It should be idempotent.

Reply:
I think we should do this in a separate pr.